test/methods/validate-fixture-input.test.ts

```diff
@@ -471,9 +593,126 @@ describe("validateFixtureInput", () => {
 
       expect(result.errors).toHaveLength(0);
     });
+
+    it("allows fields in different order than query", () => {
+      const queryAST = parse(`
+        query {
+          data {
+            items {
+              id
+              count
+              details {
+                id
+                name
+              }
+            }
+          }
+        }
+      `);
+
+      // Fixture has fields in different order: count before id, name before id
+      const fixtureInput = {
+        data: {
+          items: [
+            {
+              count: 5,
+              id: "gid://test/Item/1",
+              details: {
+                name: "Test Item",
+                id: "gid://test/ItemDetails/1"
+              }
+            }
+          ]
+        }
+      };
+
+      const result = validateFixtureInput(queryAST, schema, fixtureInput);
+
+      // Order shouldn't matter - we access by key name, not position
+      expect(result.errors).toHaveLength(0);
+    });
   });
 
   describe("Invalid Fixtures", () => {
+    it("should detect null in non-nullable scalar field", () => {
+      const queryAST = parse(`
+        query {
+          data {
+            items {
+              id
+              count
+            }
+          }
+        }
+      `);
+
+      const fixtureInput = {
+        data: {
+          items: [
+            { id: "1", count: null }
+          ]
+        }
+      };
+
+      const result = validateFixtureInput(queryAST, schema, fixtureInput);
+
+      // count is Int! so null should not be allowed
+      expect(result.errors.length).toBeGreaterThan(0);
```

Comment:
I don't think this assertion alone tells us that the test is doing what we want it to. I think we should be asserting on 1 error and checking the message matches what we expect